docs/reference/aggregations/metrics/scripted-metric-aggregation.asciidoc

```diff
@@ -26,6 +26,7 @@ POST ledger/_search?size=0
 --------------------------------------------------
 // CONSOLE
 // TEST[setup:ledger]
+// TEST[warning:params._agg/_aggs for scripted metric aggregations are deprecated, use state/states (not in params) instead. Set system property es.aggregations.enableDeprecatedScriptedMetricAggParam = false to disable this deprecated behavior.]
```

Comment:
sorry for not seeing this earlier: I think we should make the test cluster set the system property so the deprecated behaviour is not used?

Reply:
Yes it should be set to the new behavior in all tests, and a specific test added for the bwc behavior. See https://github.com/elastic/elasticsearch/pull/30975/files#diff-407c748a710cb1e33230c1fd06da7f16R694 (although adding the sysprop in BuildPlugin should really have a comment indicating the purpose and when it can be removed).

Reply:
Excellent -- thanks for the feedback. I already have a commit in another branch updating the various tests to use the new context variables (PR to follow this one) so I can bring over those changes and add another dedicated test class for the old behavior similar to `ScriptDocValuesMissingV6BehaviourTests` from that example.

I should be able to get back to this some evening this week. In the meantime to make sure we're on the same page, please let me know if this is different from what you had in mind:
- The deprecated behavior will be disabled in general via setting `enableDeprecatedScriptedMetricAggParam` to false centrally in BuildPlugin.groovy. With a comment that this gets remove presumably in 7.0.
- The deprecated behavior will be enabled again for a single dedicated integration test class (because this will get us better coverage across the affected classes vs a unit test).


Some of the test changes are substantial (ScriptedMetricIT specifically) so heads up this review will get bigger.